### PR TITLE
Create dotnet testrunner

### PR DESCRIPTION
**Pull request by srini85**
@brendanburns - this is my PR for thoughts around #75 
Appreciate your thoughts on this

### CODE REVIEW

**Comment by brendandburns** [comment]
LGTM, needs a rebase, but then ready to merge...

**Comment by srini85** [comment]
@brendanburns  - thanks, fixed conflicts

**Comment by srini85** [comment]
@brendandburns - any thoughts or suggestions on getting this one merged?

**Comment by brendandburns** [comment]
Sorry, I wanted to take a deeper look at this. I need some time to digest, appologies for the delay.

--brendan



________________________________________

**Comment by srini85** [comment]
Thanks mate.. appreciate it

File: dotnet/Metaparticle.Package/Metaparticle.cs [hunk 1]
```diff
@@ -138,6 +141,16 @@ public void Build(string[] args)
             exec.Logs(id, Console.Out, Console.Error);
         }
 
+        private void RunTests()
+        {
+            var testsAsCsv = System.Environment.GetEnvironmentVariable("METAPARTICLE_TESTS_CSV");
```
Comment (brendandburns) [inline dotnet/Metaparticle.Package/Metaparticle.cs]: Rather than do this, can we use a `[Tests ... ]` annotation?

**Comment by brendandburns** [comment]
Sorry for the delay in reviewing. I think this is great, but I would prefer this manifest as an attribute in the code, rather than an environment variable.

So it would be something like:

```cs
[Tests.Config names=["foo", "bar"]]
[Runtime.Config ...]
public void main(...) {
...
}
```

Just like [Runtime.Config](https://github.com/metaparticle-io/package/blob/master/dotnet/Metaparticle.Runtime/Config.cs) and [Package.Config](https://github.com/metaparticle-io/package/blob/master/dotnet/Metaparticle.Package/Config.cs) are attributes.

**Comment by brendandburns** [comment]
If you switch to an Attribute, I think this is ready to merge. (again, apologies for the delay in responding...)

**Comment by srini85** [comment]
Thanks @brendandburns - no issues with the delay. Will try to get that done this week.

**Comment by srini85** [comment]
@brendandburns - sorry for delay but finally got around to making the tests configuration attribute based. Thanks for suggesting a really clean approach!

**Comment by srini85** [comment]
@brendandburns any luck with this one? Let me know if you need any changes made

**Comment by brendandburns** [comment]
LGTM, thanks! (and apologies for the delay...)

**Pull request merged**